6_isSubtree.java

```diff
@@ -0,0 +1,59 @@
+/*
+MICP Summer 2017
+Author: Cindy Chen
+Language: Java
+*/
+/*
+Subtree of Another Tree
+Given two non-empty binary trees s and t, check whether tree t has exactly the same structure and node values with a subtree of s. A subtree of s is a tree that consists of a node in s and all of that node's descendants.
+*/
+/*
+1. Talk
+Will the given two trees be null? Y
+Is this a BST? N
+Check if the values are the same or the nodes are the same? Values
+
+2. Example
+See in the following pics
+
+3. Brute Force
+	1. Serialize given two trees, get two strings s and t.
+	2. Check if s contains t as a substring? if yes, return true; else return false;
```

Comment:
This won't necessarily work. Imagine you have a tree where in order the string would be "13 9 14 5 12 7 4". Now imagine you have a subtree where in order the string would be " 14 5 12". As you can see the string of the subtree is a substring of the larger one, but the structure is not the same. In the smaller tree 14 is the left, 5 is root, 12 is right. In the larger string 5 is the root but 14 is not the left and 12 is not the right. 9 is left and 7 is right.